src/sys/signal.rs

```diff
@@ -739,6 +782,15 @@ mod tests {
         assert_eq!(action_ign.handler(), SigHandler::SigIgn);
     }
 
+    #[test]
+    fn test_signal() {
+        unsafe {
```

Comment:
To prevent races against other tests, please add to the top of this function:
```rust
let m = ::SIGNAL_MTX.lock().expect("Mutex got poisoned by another test");
```

Reply:
I added the mutex guard and moved the test to `test/sys/test_signal.rs` (given it's an integration test and to access `SIGNAL_MTX`)